src/easydmp/plan/models.py

```diff
@@ -552,6 +509,114 @@ def lock(self, user, timestamp=None, wait_to_save=False):
         if not wait_to_save:
             self.save()
 
+    @transaction.atomic
+    def publish(self, user, timestamp=None):
+        if self.valid:
+            timestamp = timestamp if timestamp else tznow()
+            if not self.locked:
+                self.lock(user, timestamp, True)
+            self.generated_html = self.generate_html()
+            self.published = timestamp
+            self.published_by = user
+            self.save()
+            template = '{timestamp} {actor} published {target}'
+            log_event(user, 'publish', target=self, timestamp=timestamp,
+                      template=template)
+
+    # Template traversal
+
+    def get_first_question(self):
+        return self.template.first_question
+
+    # Validation
```

Comment:
Maybe a separate class PlanValidator could be an idea, Plan is big.

Reply:
Yeah, Plan will be much smaller once repeatable sections are in. No `data`, `previous_data` since that will be on either `AnswerSet` or removed all together by having an `Answer`-class.

I hope to be able to make these very tight single commit PR's for much of repeatable sections or we'll wind up having to add more instructions to UPGRADING I suspect.